internal/controller/nodetemplate_controller_test.go

```diff
@@ -0,0 +1,800 @@
+			By("creating Node")
+			nodes := []corev1.Node{
+				newNode("node1").withLabel(map[string]string{"node-role.kubernetes.io/worker": "true"}).withSpareTaint().build(),
+				newNode("node2").withLabel(map[string]string{"node-role.kubernetes.io/worker": "true"}).withSpareTaint().build(),
+				newNode("node3").withLabel(map[string]string{"node-role.kubernetes.io/worker": "true"}).withSpareTaint().build(),
+			}
+			for _, node := range nodes {
+				err := k8sClient.Create(ctx, &node)
+				Expect(err).ToNot(HaveOccurred())
+			}
+			By("creating NodeTemplate")
+			nodeTemplate := &nyallocatorv1.NodeTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Spec: nyallocatorv1.NodeTemplateSpec{
+					DryRun:   false,
+					Priority: 1,
+					Nodes:    2,
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"node-role.kubernetes.io/worker": "true",
+						},
+					},
+					Template: nyallocatorv1.NodeConfiguration{
+						Metadata: nyallocatorv1.NodeConfigurationMetadata{
+							Labels: map[string]string{
+								"test-label": "foo",
+							},
+							Annotations: map[string]string{
+								"test-annotation": "bar",
+							},
+						},
+						Spec: nyallocatorv1.NodeConfigurationSpec{
+							Taints: []corev1.Taint{
+								{
+									Key:    "test-taint",
+									Value:  "baz",
+									Effect: corev1.TaintEffectNoSchedule,
+								},
+							},
+						},
+					},
+				},
+			}
+			err := k8sClient.Create(ctx, nodeTemplate)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("checking NodeTemplate status")
+			checkNodeTemplateStatus("test", true)
+
+			By("checking node is properly configured")
+			n := &corev1.NodeList{}
+			err = k8sClient.List(ctx, n, client.MatchingLabels{"nyallocator.cybozu.io/node-template": "test"})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(n.Items).To(HaveLen(2), "should have created 2 nodes")
+			for _, node := range n.Items {
+				Expect(node.Labels).To(HaveKeyWithValue("test-label", "foo"), "node should have the correct label")
+				Expect(node.Annotations).To(HaveKeyWithValue("test-annotation", "bar"), "node should have the correct annotation")
+				Expect(node.Spec.Taints).To(ContainElement(corev1.Taint{
+					Key:    "test-taint",
+					Value:  "baz",
+					Effect: corev1.TaintEffectNoSchedule,
+				}), "node should have the correct taint")
+			}
+
+			By("checking metrics are exposed correctly")
+			resp, err := http.Get("http://localhost:8080/metrics")
+			Expect(err).ToNot(HaveOccurred())
+			defer resp.Body.Close()
+			body, err := io.ReadAll(resp.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(string(body)).To(ContainSubstring("nyallocator_expected{name=\"test\"} 2"))
+			Expect(string(body)).To(ContainSubstring("nyallocator_current{name=\"test\"} 2"))
+			Expect(string(body)).To(ContainSubstring("nyallocator_sufficient{name=\"test\"} 1"))
+		})
+
+		It("should not create nodes when dry run is enabled", func() {
```

Comment:
allocate?
https://github.com/cybozu-go/nyallocator/blob/47365fc009d801d12e86295984d7b8af40f6ee88/docs/design.md?plain=1#L23C15-L23C25
```suggestion
		It("should not allocate nodes when dry run is enabled", func() {
```